packages/backend-common/README.md

```diff
@@ -0,0 +1,36 @@
+# @backstage/backend-common
+
+Common functionality library for Backstage backends, implementing logging,
+error handling and similar.
+
+## Usage
+
+Add the library to your backend package:
+
+```sh
+yarn add @backstage/backend-common
+```
+
+then make use of the handlers and logger as necessary:
+
+```typescript
+import {
+  logger,
+  errorHandler,
+  notFoundHandler,
+} from '@backstage/backend-common';
+
+const app = express();
+app.use('/home', myHomeRouter);
+app.use(errorHandler());
+app.use(notFoundHandler());
```

Comment:
Hm, how do you mean? It feels very unlikely that a backend plugin would ever want unique 404 handling for its subroutes, but if they do, they actually can do so at the end of their own "chain" AFAIK.

I feel it's pretty reasonable to have one global 404 handler per "app" (backend service in this case).

Reply:
Main goal here overall is to make it super easy to make a backend, piecing together reasonable prefab things where possible (and sure, inventing your own if you have to, but it's targeting the 80% case)

Reply:
Oh BTW to clarify if I was unclear - this is the handling of requests that try a URL that isn't matched by any plugin whatsoever. Not like, a valid inventory item URL that happened to point to a component ID that doesn't exist.

Reply:
Was mainly referring to the error handler, and yeah they can put it at the end of their chain if needed. My though was that they should put it at the end of the chain so that they don't throw an error and hope that the app catches it properly. I think this is fine though since now the app can log errors inside plugins in a way it couldn't before.

I don't think either approach is definitely more simple. In this case the person reading and writing plugin code needs to know that the app will handle errors for them. If we put them inside the plugin they need to be imported and added there. Both are pretty minor issues imo.

Reply:
I was intending for the custom response code errors to be a well known "api" for conveying http type errors, but now that I think about it - it's poor form to work like that, your own handler should be translating business layer exceptions / odd states. That essentially leaves "unintended" problems that most likely will just be 500s. That's the kind of thing that this middleware would be intended to cope with in a coherent way.